packages/sitecore-jss/src/graphql-request-client.ts

```diff
@@ -19,24 +27,20 @@ export class GraphQLRequestClient {
     query: string | DocumentNode,
     variables?: { [key: string]: unknown }
   ): Promise<T> {
-    const client = new GraphQLClient(
-      this.endpoint,
-      this.apiKey ? { headers: { sc_apikey: this.apiKey } } : undefined
-    );
-    const onError = (error: unknown) => {
-      console.error(
-        chalk.red(`
-          Error occurred while attempting to fetch graphQL data.
-          Endpoint: ${this.endpoint}
-          Query: ${query}
-          Variables: ${JSON.stringify(variables, null, 2)}
-          Error: ${JSON.stringify(error, null, 2)}
-        `)
-      );
-      throw error;
-    };
-
-    // todo: we should also print the query if we are in debug mode (Anastasiya, March 2021)
-    return await client.request(query, variables).catch(onError);
+    return new Promise((resolve, reject) => {
```

Comment:
It seems odd to create a new promise when we already have a promise returned from client.request. Why is this needed?

Reply:
This is necessary for us to be able to add our debug log hooks to the response (within `then`/`catch`).